.travis.yml

```diff
@@ -28,17 +28,17 @@ addons:
       - llvm-toolchain-precise
 
 env:
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Ninja"
-  - COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
-
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.7 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
-  - COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
-
-  - COMPILER=gcc CMAKE_GENERATOR="Ninja"
-  - COMPILER=gcc CMAKE_GENERATOR="Unix Makefiles"
+  # Test all generators and Clang + GCC
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Ninja"
+  - TEST=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
+
+  # Benchmarks
+  - BENCHMARK=true COMPILER=clang++-3.6 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.7 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=clang++-3.8 CMAKE_GENERATOR="Unix Makefiles"
+  - BENCHMARK=true COMPILER=gcc         CMAKE_GENERATOR="Unix Makefiles"
```

Comment:
> It would be useful to test a few versions so we can check whether the detection of supported standards works.

Ok, you're right.

> How do we push the results produced by the various Travis jobs to gh_pages?

I don't think that's an issue, we just have to push them to separate directories. For example, `clang-3.5/` gets all the benchmarks that were run on Clang 3.5, etc..


> Metabench would then become a tool for comparing compilers besides the original intent of comparing different implementations of metaprogramming techniques. Do we want that?

I do not think that would make Metabench a tool for comparing compilers (I certainly think this is out of scope for Metabench, in all cases). However, the idea is that metaprogramming libraries can have different performance profiles depending on the compiler, both because of internal compiler implementation and also special casing within a library implementation. For example, when some Clang intrinsics are available, Hana will compile faster than when they're not available. It could be the case that such intrinsics are only available starting at some version of the compiler.

All the trouble I'm going through to benchmark on multiple compilers stems from the assumption that it's useful to know how a metaprogramming library performs on a specific compiler, because compile-time performance is highly dependent on the compiler. Experience with Hana suggests that this is the case.